src/roles/vsd-dbbackup/tasks/main.yml

```diff
@@ -53,6 +53,31 @@
     - inventory_hostname in groups['vsds']
     - not vsd_inplace_upgrade
 
+- block:
+  - name: find pre upgrade backup script
+    include_role:
+      name: common
+      tasks_from: find-file.yml
+    vars:
+      dir_name: "{{ nuage_unzipped_files_dir }}"
+      pattern: "pre-upgrade-backup.*"
+  delegate_to: localhost
+- name: VSD pre upgrade backup script location
+  set_fact:
+    image_path: "{{ file_path }}"
+    image_file_name: "{{ file_name }}"
+- name: Copy and send preupgrade check script to vsd 
+  copy:
+    src: "{{ image_path }}/{{ image_file_name }}"
+    dest: "/opt/vsd/data/"
+  delegate_to: localhost
+  remote_user: "{{vsd_default_username}}"
+  become: True
+- name: Run silent preupgrade check script in the vsd
+  script: "/opt/vsd/data/pre-upgrade-backup.sh -s"
+  remote_user: "{{ vsd_custom_username | default(vsd_default_username) }}"
+  become: yes
+
```

Comment:
This should all be run only when the pre-upgrade script is present. Add a when condition for the block. Else it will fail when the script is not present.

Reply:
added when condition for this block